### ISSUE TITLE
Bug: switching boards shows wrong board view initially

### ISSUE THREAD

**Issue by kamre** [mattermost/focalboard#3780]
## Steps to reproduce the behavior

1. Create new empty board (name it `Kanban`)
2. Create another new empty board (name it `Gallery`)
3. Add gallery view and delete board view from the board named `Gallery` 
4. Create new card in the gallery view
5. Open card dialog and change property from `Select` to `Text`
6. Switch to `Kanban` board in the left sidebar
7. Switch back to `Gallery` board in the left sidebar
8. Empty screen and the following error in the console: `ASSERT [Board views must have groupByProperty set]`

## Expected behavior

After switching back to `Gallery` board its gallery view with single card should be visible and there should be no errors in the console.

## Edition and Platform

 - Edition: Personal Server
 - Version: `main` branch
 - Browser and OS: Chrome on Mac

## Additional context (optional)
After the error it is possible to reload the page in the browser and proper board view will be visible.

The reason for such error is that board view type is not immediately changed to the expected one, only after first rendering the proper board view type is rendered.

* Sev 2: Affects major functionality with a difficult or non-obvious workaround

**Comment by wuwinson** [comment]
Related to https://github.com/mattermost/focalboard/issues/3646 - need to handle lack of `Select` property or empty values within a `Select` property.

**Comment by wuwinson** [comment]
cc @asaadmahmood
